### PR TITLE
feature/#26-add-form-component

### LINKED ISSUE #26: Add Form component
image
image
image
image
image
image

## Related issues

- #22 
- #21 
- #25

### PR DESCRIPTION
**Pull request by AlejandroSalcido**
### Description

Form Component enhanced re usability, react context basic usage in order to handle with form inputs actions and values.

#### Type of change

- [x] New feature (non-breaking change which adds functionality)

### How to watch it:
Search into the github branches and select the one named: 'feature/#26-add-form-component', once there fetch and pull in order to update the code. Lastly use the command you use to initialize the app ant check how the form look on the home ' / ' page

### Checklist:

- [x] Doesn't break the current code.
- [x] Passes linters and test, also is building.
- [ ] Doesn't have spelling or grammatical problems.
- [ ] Doesn't have unnecessary comments or debugging code.
- [ ] The branch is updated with the last dev branch changes.

### CODE REVIEW

File: src/components/Form.jsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import styled from 'styled-components'
+import Button from '../components/Button'
+import Input from '../components/Input'
+
+const Form = () => {
```
Comment (Israel-Laguan) [inline src/components/Form.jsx]: please add props like onSubmit, action, target and validation

Reference: https://www.w3schools.com/html/html_forms_attributes.asp

File: src/components/Form.jsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      <Input placeholder="Celular"></Input>
+      <br />
+      <Button>confirmar</Button>
+    </FormWrapper>
+  )
+}
+
+export default Form
+
+const FormWrapper = styled.div`
```
Comment (Israel-Laguan) [inline src/components/Form.jsx]: should use styled.form

File: src/components/Form.jsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  width: 100vw;
+  height: 100vh;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  @media (min-width: 960px) {
+    background: yellow;
```
Comment (Israel-Laguan) [inline src/components/Form.jsx]: this style is not needed, should be none or transparent

File: src/components/Form.jsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      <Button>confirmar</Button>
+    </FormWrapper>
+  )
+}
+
+export default Form
+
+const FormWrapper = styled.div`
+  width: 100vw;
+  height: 100vh;
```
Comment (Israel-Laguan) [inline src/components/Form.jsx]: this is not good, should be 100%

File: src/components/Form.jsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+`
+
+const FormTitle = styled.h1`
+  text-transform: uppercase;
+  color: yellow;
+  font-size: 24px;
+  @media (min-width: 960px) {
+    color: white;
+  }
+`
```
Comment (Israel-Laguan) [inline src/components/Form.jsx]: this is not needed

File: src/components/Form.jsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+const Form = () => {
+  return (
+    <FormWrapper>
+      <FormTitle>confirma recepcion</FormTitle>
+      <Input placeholder="Nombre"></Input>
+      <Input placeholder="RUT"></Input>
+      <Input placeholder="Celular"></Input>
+      <br />
+      <Button>confirmar</Button>
```
Comment (Israel-Laguan) [inline src/components/Form.jsx]: this is not needed

File: src/components/Form.jsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
+import styled from 'styled-components'
+import Button from '../components/Button'
+import Input from '../components/Input'
```
Comment (Israel-Laguan) [inline src/components/Form.jsx]: this is not needed

**Review by Israel-Laguan — CHANGES_REQUESTED** [review]
Please remade this whole component. Remember this should be reusable and composable.

Also add an example of use.

File: src/components/Form.jsx (hunk 4 above)
Comment (AlejandroSalcido) [reply src/components/Form.jsx]: Corrected

File: src/components/Form.jsx (hunk 3 above)
Comment (AlejandroSalcido) [reply src/components/Form.jsx]: Done

File: src/components/Form.jsx (hunk 5 above)
Comment (AlejandroSalcido) [reply src/components/Form.jsx]: minimal fixes done

File: src/components/Form.jsx (hunk 6 above)
Comment (AlejandroSalcido) [reply src/components/Form.jsx]: So form would be only a list of multiple input, no button?

File: src/components/Form.jsx (hunk 2 above)
Comment (AlejandroSalcido) [reply src/components/Form.jsx]: div changed to a form!

File: src/components/Form.jsx (hunk 7 above)
Comment (AlejandroSalcido) [reply src/components/Form.jsx]: if input is not going to be recycled here, form would be a 3 input reusable component, or how we are going to hadle it?

File: src/components/Form.jsx (hunk 1 above)
Comment (AlejandroSalcido) [reply src/components/Form.jsx]: hadeled submit inside the form

File: src/components/Form.jsx (hunk 6 above)
Comment (Israel-Laguan) [reply src/components/Form.jsx]: Maybe the buttons can be part of the form component. About Input we can just make them children in this case. we can check this case in a meeting later to clarify.

File: src/pages/Home.jsx [hunk 8]
```diff
@@ -12,7 +13,17 @@ const Home = () => (
     <Text as="button" bold uppercase>
       Aceptar
     </Text>
-    <Form />
+    <Form>
```
Comment (Israel-Laguan) [inline src/pages/Home.jsx]: You forgot a onSubmit event here, also manage errors, but is OK for now.

Just include onSubmit and we can merge

File: src/components/Form.jsx [hunk 9]
```diff
@@ -1,8 +1,17 @@
...
+import React, { useState } from 'react'
+import PropTypes from 'prop-types'
+
+export const FormContext = React.createContext({
+  form: {},
+  handleFormChange: () => {},
+})
+
+const Form = props => {
+  const { children } = props
```
Comment (Israel-Laguan) [inline src/components/Form.jsx]: this is not needed, move this to line 10, like:

`const Form =({ children, onSubmit }) => {`

File: src/components/Form.jsx [hunk 10]
```diff
@@ -14,22 +23,26 @@ const Form = () => {
...
+        {children}
+      </FormContext.Provider>
+
       <FormButton onclick={handleSubmit}>Confirmar</FormButton>
     </FormContainer>
   )
 }
 
+Form.propTypes = {
+  children: PropTypes.any,
```
Comment (Israel-Laguan) [inline src/components/Form.jsx]: Please avoid using any. In this case for example use `PropTypes.node`

**Review by Israel-Laguan — CHANGES_REQUESTED** [review]
Good job with the provider method! Just few things and we can merge

**Comment by Israel-Laguan** [comment]
Please rebase

**Comment by Israel-Laguan** [comment]
If I want to get data from a parent component I notice there is no way

in Check I was trying to get the value of packageCode before checking with modal, but your implementation makes it difficult. Please explore other way

**Comment by denisosuna** [comment]
> If I want to get data from a parent component I notice there is no way
> 
> in Check I was trying to get the value of packageCode before checking with modal, but your implementation makes it difficult. Please explore other way

I have made some changes to solve this, I think it works fine, please check

**Comment by Israel-Laguan** [comment]
I see this error on console 
image

**Comment by denisosuna** [comment]
> I see this error on console image

this error should be solve, please check @Israel-Laguan

**Review by Israel-Laguan — APPROVED** [review]

**Comment by Israel-Laguan** [comment]
Tested again and worked without the error I found last time

**Pull request merged**